java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/model/GenericMqttMetadata.java

```diff
@@ -0,0 +1,66 @@
+package com.cumulocity.generic.mqtt.client.model;
+
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+
+/**
+ * Metadata for {@link GenericMqttMessage}.
+ */
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class GenericMqttMetadata {
+
+    /**
```

Comment:
question: is this the client that is receiving it? If I have questions, less informed users will too.

The reason I ask is I wonder why is this present. I assume it is used server side for message routing to this client. Should the server be removing this? It is a server side client-id? Is it the (pulsar) subscriber name. Clients would not think they have an id. This sounds like server implementation detail to me, but I might be wrong

Reply:
It's the MQTT client ID. MQTT clients do add a client ID.